File: Novels/packages/5-mobile-ios/StoryReader/Model/ReadingProgress.swift
import Foundation

// Reading Progress model - CRITICAL for cross-device sync
struct ReadingProgress: Codable {
    let id: String?
    let userId: String
    let storyId: String
    let chapterId: String
    let position: Double // 0.0 - 1.0 (scroll position or page number)
    let completedAt: Date?
    let syncedAt: Date?
    let syncStatus: SyncStatus
    
    enum SyncStatus: String, Codable {
        case synced
        case pending
        case conflict
        case error
    }
    
    enum CodingKeys: String, CodingKey {
        case id
        case userId = "user_id"
        case storyId = "story_id"
        case chapterId = "chapter_id"
        case position
        case completedAt = "completed_at"
        case syncedAt = "synced_at"
        case syncStatus = "sync_status"
    }
}

File: Novels/packages/6-mobile-android/StoryReader/app/src/main/java/com/storysphere/storyreader/model/ReadingProgress.kt
package com.storysphere.storyreader.model

data class ReadingProgress(
    val id: String,
    val userId: String,
    val storyId: String,
    val chapterId: String,
    val position: Int = 0, // Scroll position or page number
    val progress: Float = 0f, // 0.0 to 1.0
    val wordsPerMinute: Int? = null,
    val readingTime: Long = 0L, // milliseconds
    val lastReadAt: Long = System.currentTimeMillis(),
    val syncStatus: SyncStatus = SyncStatus.SYNCED,
    val lastSyncedAt: Long? = null
)

